Lizard Food: a Taste of the Homeworld

EOBGames opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Adds a wide selection of cultural food and drinks for lizards, in the form of new chef recipes, new drinks and a new botanical plant: the Korta Nut. (Now featuring far better sprites by @AdipemDragon!)
![lizardcuisine2](https://user-images.githubusercontent.com/58124831/120835990-57c79b00-c565-11eb-8541-ed7d95bd326a.png)
The aims of this PR are threefold:
1. To introduce lizard culture (based upon Common Core) in a tangible way.
2. Give some more variety of techniques and recipes to the chef and bartender.
3. To give some flavour (metaphorically and literally).

A complete list of recipes can be found here, so as to not clog up the PR body: 
I will, however, give special mention to some of the recipes as they may be confusing:
Nizaya pasta is made by slicing a rootdough slice, not in the processor like normal pasta.
Root flatbreads are made by grilling a flat rootdough.
Both blood sausage and headcheese must be cured on a drying rack before they're ready.
You can also find a few of the ingredients, namely the canned ones, moonfish eggs, vinegar, and quality oil, at the chef's produce console.

New Drinks
The chef isn't the only one getting love here!
![bar_drinks](https://user-images.githubusercontent.com/58124831/120428736-80d50980-c374-11eb-8a04-06883e873efb.PNG)
Some new bar drinks are also available, using two new lizardy drinks: Kortara, the product of fermenting Korta Nuts, and Mushroom Tea, made from Lavaland mushrooms (specifically the ones that give shavings). You can even turn your Mushroom Tea into delicious Mushi Kombucha!
Recipes can be seen here: https://pastebin.com/g2JXFcEA

Korta Nuts
![korta_nut](https://user-images.githubusercontent.com/58124831/120223265-a281a400-c241-11eb-8d9e-268509096ae9.PNG)
Korta Nuts are a versatile tool in lizard cuisine, with two key ingredients coming from them: korta flour, made by grinding them, and korta milk, made by juicing them. They can also be mutated into Sweet Korta Nuts, which produce korta nectar: a sweet syrup used in a few dishes. They can be grown at botany, with the seeds being located in the Megaseed Vendor.

New Aquarium Fish
4 new aquarium fish are also included, direct from Tizira's oceans to your glass prison:
![feesh](https://user-images.githubusercontent.com/58124831/120224006-da3d1b80-c242-11eb-926e-60f9e104ea3c.png)
The dwarf moonfish, gunner jellyfish, needlefish and armorfish make for excellent pets... or excellent meals.

Other minor things introduced with this PR include:
- The NUTS foodtype, currently used only for peanuts and korta nuts, but it is expandable with new features later.
- Peanuts and "C&Ds" (we cannot legally say they won't melt in your hand) at the snack vendor.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
We are critically lacking in tangible links to the Common Core lore in game right now, and I think this is a good first step in bringing it into the game.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl: Inept, Coiax, AdipemDragon, YakumoChen
add: The release of the new cookbook, "Tiziran Cooking: a Taste of the Homeworld" has brought Lizard food to the masses! Try out some new treats, like Moonfish, Nectar Larvae, and perhaps even Headcheese at the kitchen today!
add: To coincide with the new popularity of Lizard cuisine, Nanotrasen now stocks seeds for Korta Nuts, a common ingredient in lizard cooking, in the Megaseed vendors. Botanists, you know what to do.
add: A few Tiziran fish are also available for you aquaculture lovers out there. They all like saltwater!
add: A few new snacks are also now available at the station's snack vendors, for those of you too lazy to visit the kitchen.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

EOBGames committed:
by the nine divines

EOBGames committed:
Cease and Desist

EOBGames committed:
i have zero fucks left to give

EOBGames committed:
that should do it

EOBGames committed:
unfucks icons

Fikou commented on code/game/objects/items/food/snacks.dm:
```diff
@@ -168,3 +168,79 @@
+	switch(peanut_flavour)
+		if("salted")
+			name = "\improper Parker's salted peanuts"
+			desc = "As commonly seen on shuttle flights. Tastes salty, just like most peanuts."
```
cant you just add the second part to the original desc?

Fikou commented on code/game/objects/items/food/snacks.dm:
```diff
@@ -168,3 +168,79 @@
+			name = "\improper Parker's salted peanuts"
+			desc = "As commonly seen on shuttle flights. Tastes salty, just like most peanuts."
+			food_reagents = list(/datum/reagent/consumable/nutriment = 2, /datum/reagent/consumable/salt = 1)
+			tastes = list("peanuts" = 2, "salt" = 1)
```
same with here

Fikou commented on code/game/objects/items/food/snacks.dm:
```diff
@@ -168,3 +168,79 @@
+		if("salted")
+			name = "\improper Parker's salted peanuts"
+			desc = "As commonly seen on shuttle flights. Tastes salty, just like most peanuts."
+			food_reagents = list(/datum/reagent/consumable/nutriment = 2, /datum/reagent/consumable/salt = 1)
```
and here, im not even sure this will work if you change food_reagents after initialize?

EOBGames commented:
I'm now seeing that botany sprites do not work the way I wish they worked. I'll correct that sometime tonight.

coiax commented on code/game/objects/items/food/snacks.dm:
```diff
@@ -168,3 +168,79 @@
+	name = "\improper Parker's peanuts"
+	desc = "As commonly seen on shuttle flights."
+
+/obj/item/food/peanuts/flavoured/Initialize()
```
I would probably make each of these as subtypes, and add a `random` subtype that picks one of the other types, and then reassigns the desc, flavour, and reagents?

Not sure. It is only peanuts.

coiax commented on code/modules/food_and_drinks/food/condiment.dm:
```diff
@@ -254,6 +254,20 @@
 
+/obj/item/reagent_containers/food/condiment/vinegar
+	name = "vinegar"
+	desc = "Perfect for chips, if you're feeling British."
```
```suggestion
	desc = "Perfect for chips, if you're feeling Space British."
```

coiax commented on code/game/objects/items/food/lizard.dm:
```diff
@@ -0,0 +1,564 @@
+	desc = "A popular Tiziran streetfood consisting of vinegar-marinated armorfish on a skewer with onion and chillis."
+	icon = 'icons/obj/food/lizard.dmi'
+	icon_state = "picoss_skewer"
+	food_reagents = list(/datum/reagent/consumable/nutriment/protein = 3, /datum/reagent/consumable/nutriment = 3, /datum/reagent/consumable/vinegar= 1, /datum/reagent/consumable/capsaicin = 1)
```
Long enough that you can split this into multiple lines.

EOBGames replied:
I'm not saying I wouldn't do it- but I don't think any other food splits over more than one line. Would it not be better to just leave this as is?

coiax commented on code/modules/food_and_drinks/recipes/tablecraft/recipes_lizard.dm:
```diff
@@ -0,0 +1,389 @@
+		/datum/reagent/consumable/salt = 2
+	)
+	result = /obj/item/food/raw_tiziran_sausage
+	subcategory = CAT_LIZARD
```
Has the code that means only lizard chefs can cook this already been added? Or is this common pool recipe.

EOBGames replied:
These are all common pool since attempting to make it lizard only nearly drove me mad last night. If any miracle solutions present themselves I'd be happy to make it lizard only (with an accompanying cookbook for non-lizards).

EOBGames committed:
korta nut bush icons

EOBGames committed:
code is important too

YakumoChen commented:
I thought lizards were carnivores, why does their cuisine include a bunch of pastries?

EOBGames commented:
> I thought lizards were carnivores, why does their cuisine include a bunch of pastries?

Lizards specifically like meat, but are omnivorous. The only foods they cannot eat are grain and dairy, of which neither is used in any of these recipes- the breads and such are made from vegetables and nuts.

RaveRadbury commented on code/game/objects/items/food/lizard.dm:
```diff
@@ -0,0 +1,564 @@
+	icon_state = "lung_stirfry"
+	food_reagents = list(/datum/reagent/consumable/nutriment/protein = 3, /datum/reagent/consumable/nutriment/vitamin = 2, /datum/reagent/consumable/capsaicin = 1)
+	tastes = list("meat" = 1, "heat" = 1, "veggies" = 1)
+	foodtypes = MEAT | VEGETABLES
```
```suggestion
	foodtypes = MEAT | VEGETABLES|GROSS
```
"Delicious, if you like lungs."

coiax replied:
Why do you hate spaces between things Rave 😛

RaveRadbury replied:
Idonotknowwhatyoumeancoiax

thgvr commented:
Very good and fun to play around with, people's initial reactions were also very interesting. 
![image](https://user-images.githubusercontent.com/81882910/120249756-c092d880-c230-11eb-9bfa-d9700f7f91cf.png)
Hope to see more lore-flavor-feature things in the future.

EOBGames commented:
> Very good and fun to play around with, people's initial reactions were also very interesting.
> ![image](https://user-images.githubusercontent.com/81882910/120249756-c092d880-c230-11eb-9bfa-d9700f7f91cf.png)
> Hope to see more lore-flavor-feature things in the future.

Don't worry, I have more ideas up in the grey matter.

EOBGames committed:
fixes some bugs, adds bar content

now go get drunk!

EOBGames committed:
oops!

Mothblocks commented:
Drafting as per @coiax 's feedback.

ATH1909 commented on code/modules/hydroponics/grown/korta_nut.dm:
```diff
@@ -0,0 +1,53 @@
+	maturation = 10
+	production = 10
+	mutatelist = list()
+	reagents_add = list(/datum/reagent/consumable/korta_nectar, /datum/reagent/consumable/nutriment/vitamin = 0.04, /datum/reagent/consumable/nutriment = 0.1)
```
should "/datum/reagent/consumable/korta_nectar" have an "= [number]" next to it?

ATH1909 commented on code/modules/reagents/chemistry/reagents/drink_reagents.dm:
```diff
@@ -1026,3 +1026,19 @@
+
+/datum/reagent/consumable/mushroom_tea/on_mob_life(mob/living/carbon/M, delta_time, times_fired)
+	if(islizard(M))
+		M.adjustOxyLoss(-0.5 * REM * delta_time, 0)
```
remember that REM is 0.5, so this will heal a pathetic 0.25 points of oxyloss per second.

EOBGames replied:
yes

RaveRadbury commented on code/modules/food_and_drinks/recipes/drinks_recipes.dm:
```diff
@@ -655,3 +655,33 @@
+
+/datum/chemical_reaction/drink/protein_blend
+	results = list(/datum/reagent/consumable/ethanol/protein_blend = 5)
+	required_reagents = list(/datum/reagent/yuck = 1, /datum/reagent/consumable/korta_flour = 1, /datum/reagent/blood = 1, /datum/reagent/consumable/ethanol = 2)
```
```suggestion
	required_reagents = list(/datum/reagent/yuck = 1, /datum/reagent/consumable/korta_flour = 1, /datum/reagent/blood = 1)
```
I'm gonna keep saying that a protein drink shouldn't have ethanol in it.

EOBGames replied:
listen, if you were gonna drink a blended mouse you'd want to be getting drunk off it too

RaveRadbury replied:
not true, lizards enjoy eating mice

EOBGames replied:
and I enjoy eating burgers, but blend one up and suddenly I'm not so keen

Mothblocks replied:
we'll see about that

EOBGames committed:
you want this mushroom?

right, this first one's based on... I was in a pub, a few months ago having a breakfast, right? this big fry up breakfast... and on the plate was a mushroom, big fried mushroom, right, and I don't like mushrooms so I didn't eat it, but, I thought "d'you know, that's a waste, this mushroom just going to waste, it would be good if I could go up to one of the other tables and say, just approach em and just go like "excuse me, d'you want this mushroom?" and I just thought of an idea for a character that does that, just goes like that all the time, just goes up to people, like "scuse me, d'you want this mushroom?" "d'you want this mushroom?" just goes up with this big gigantic plate with this big mushroom at the edge of it "d'you want this mushroom?" but the thing is he's no like, he's no just like a guy in a pub, he could be, like anywhere, turn up anywhere, at any time, just like really inappropriate and really funny, inappropriate situations, like a... imagine like a funeral, so he turns up at a funeral so it would be like, everybody would be like that "aaaaaaaaammmmmmmeeeeeennnnnn" and then you hear this "d'you want this mushroom?" "d'you want this mushroom?" just picture it, like a huge, huge mushroom at the... aye, just forget it.

EOBGames committed:
god

EOBGames committed:
heresiarch is dead, drunken espatier might work

EOBGames committed:
..()

coiax committed:
Rework random peanuts and candy to work in place (#20)

Instead of creating a new item, we instead instance a temporary "flavour
sample", then copy over the name, description, flavours and reagents
before the rest of the food Initializes.

EOBGames commented:
By the way @coiax I'm awaiting some sprites, so this PR isn't ready for merge quite yet even if the code is all good (which I'm not quite 100% sure on yet anyway).

Mothblocks commented:
@coiax  Are you sure its better to do it in place? What if we want to attach some other behavior to a specific set of peanuts/C&Ds?

I think it's better to just make `/obj/item/peanuts_spawner` or whatever, and revert it to the code it was before.

coiax commented:
@Mothblocks 

It seemed easier to modify in-place some items that had no other special behaviour, than adding spawner support, or "random items with a single category" support for vending machines.

There wasn't a test or anything I was trying to pass, just trying to make sure that when the item was vended, it went into the person's hand directly, rather than being spawned elsewhere?

Mothblocks commented:
I see, guess we'll just kick that can down the line.

coiax commented:
> 
> 
> I see, guess we'll just kick that can down the line.

Nah, I don't want to be doing that. I have a possible solution that I was thinking about last night.

Basically, let me have a look at the vending machine code, which I believe has a simple associated list for
type -> amount, for things it vends. I think we could easily make a "composite entry" datum type to go
into the type, where you have multiple potential items represented by a single type.

I'll see if I can make a PR today to this effect.

EOBGames committed:
food from the dragon

thanks adipem

EOBGames committed:
steeped shrooms

EOBGames committed:
seraka mycelium

EOBGames commented:
So I got the new sprites and gave everything a once over to make sure everything was working- it all seems to be, so I'd say this is ready on my end. @coiax are you working on vending stuff for this PR or are you doing it separately?

YakumoChen commented:
Missed opportunity to call the peanuts "Salt PRs" (Salted Peanut Roasts) and make various flavors based on salt PRs. 

"Contains 30 reasons you should REMOVE HUNGER NOW!!!"

YakumoChen commented on code/modules/aquarium/fish.dm:
```diff
@@ -163,3 +163,36 @@
+/datum/aquarium_behaviour/fish/armorfish
+	name = "armorfish"
+	desc = "A small shellfish native to Tizira's oceans, known for its exceptionally hard shell. Consumed similarly to prawns."
+	icon_state = "needlefish"
```
wrong icon_state

EOBGames committed:
peanuts

YakumoChen commented:
>Peanut gallery

Laughing so hard and wondering how I didn't think of this instead of Salt PRs

YakumoChen commented:
@EOBGames did you see the PR I made on your base btw? Regular food in the snack vendors should have junkiness values and the candy should probably be made of mostly sugar instead of nutriment.

PKPenguin321 commented:
Don't worry I will add a "peanut post" recipe that is a nut on a kebab stick later

EOBGames committed:
sugar and junk

thanks to yakumochen

EOBGames commented:
> @EOBGames did you see the PR I made on your base btw? Regular food in the snack vendors should have junkiness values and the candy should probably be made of mostly sugar instead of nutriment.

alright, I've implemented most of the changes from that PR (I wanted to make some alterations so figured it was best making them myself and crediting you), thanks
